File: src/ObjectManager/Object.Tes/FilePacks/Records/040-ROAD.Road.cs
using OA.Core;

namespace OA.Tes.FilePacks.Records
{
    public class ROADRecord : Record
    {
        public override string ToString() => $"ROAD:";
        public PGRDRecord.PGRPField[] PGRPs { get; set; }
        public UNKNField PGRR { get; set; }

        public override bool CreateField(UnityBinaryReader r, GameFormatId format, string type, int dataSize)
        {
            switch (type)
            {
                case "PGRP":
                    PGRPs = new PGRDRecord.PGRPField[dataSize >> 4];
                    for (var i = 0; i < PGRPs.Length; i++) PGRPs[i] = new PGRDRecord.PGRPField(r, dataSize); return true;
                case "PGRR": PGRR = new UNKNField(r, dataSize); return true;
                default: return false;
            }
        }
    }
}

File: src.mac/ObjectManager/ObjectManager/Object.Tes/FilePacks/Records/040-ROAD.Road.swift
//
//  ROADRecord.swift
//  ObjectManager
//
//  Created by Sky Morey on 5/28/18.
//  Copyright © 2018 Sky Morey. All rights reserved.
//

public class ROADRecord: Record {
    public override var description: String { return "ROAD:" }
    public var PGRPs: [PGRDRecord.PGRPField]!
    public var PGRR: UNKNField!
    
    override func createField(_ r: BinaryReader, for format: GameFormatId, type: String, dataSize: Int) -> Bool {
        switch type {
        case "PGRP": PGRPs = r.readTArray(dataSize, count: dataSize >> 4)
        case "PGRR": PGRR = r.readBYTV(dataSize)
        default: return false
        }
        return true
    }
}
